### ISSUE TITLE
Login  with whitespace username issue- > 3.17.1 client

### ISSUE THREAD

**Issue by LaCruz75** [nextcloud/android#9194]
I have issue with Login users with whitespace username via Android App.
I tried latest Andoird App (3.17.. and 3.18RC2) without success.
I rollback to Andorid Client 3.13.1 and I can login with usernames with whitepaces.

On success login with 3.13.1 an upgrade to 3.18+ it shows "server not avaible"

_Originally posted by @LaCruz75 in https://github.com/nextcloud/android/issues/3623#issuecomment-958703768_

**Comment by AlvaroBrey** [comment]
Hey @LaCruz75, please fill out the following information:
```
Environment data

Android version:

Device model:

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

Reverse proxy:
```

Additionally, if you can, please test to see if [version 3.17.0](https://github.com/nextcloud/android/releases/tag/stable-3.17.0) works for you, to figure out if there was a regression due to changes in .1.

**Comment by LaCruz75** [comment]
Environment data

Android version: 11

Device model: Samsung S10

Stock or customized system: Stock

Nextcloud app version: 
working 3.13.1, 3.17.0, 3.17.1
****issue**: since 3.18.0 RC1**

Nextcloud server version: 22.2.0

Reverse proxy: Sophos UTM9

**Comment by LaCruz75** [comment]
Since 3.18.0 RC1 I got the issue. It also shows on upgraded account "Server nicht verfügbar"/"Server not available"

**Comment by fl-ma** [comment]
Can confirm this also occurs for usernames that contain pipe (e.g. due to using SAML/SSO): auth0|abcd
Seems to have worked fine in earlier releases (but no details about that available unfortunately).

Problem does not show when using a "regular" username: abcd

Android version: 12
Device model: Pixel 4A
Stock or customized system: Stock
Nextcloud app version: 3.18.0
Nextcloud server version: 20.0.14

**Comment by AlvaroBrey** [comment]
Whitespace in usernames should be fixed by https://github.com/nextcloud/android-library/pull/756 and https://github.com/nextcloud/android-library/pull/761, which will be  released in the upcoming 3.18.1 version, so I'm closing this issue.

@fl-ma if you want, you can test the latest dev version, which should include the above PRs and see if they fix pipes too.

**Comment by LaCruz75** [comment]
> Whitespace in usernames should be fixed by [nextcloud/android-library#756](https://github.com/nextcloud/android-library/pull/756) and [nextcloud/android-library#761](https://github.com/nextcloud/android-library/pull/761), which will be released in the upcoming 3.18.1 version, so I'm closing this issue.
> 
> @fl-ma if you want, you can test the latest dev version, which should include the above PRs and see if they fix pipes too.

Finally wokrs with latest.apk (202111201)
Thanks
